File: snapshots/output-net10.0/syntax/Main/Interfaces.cs
  using System.Diagnostics.CodeAnalysis;
//      ^^^^^^ read_access scip-dotnet nuget . . System/
//             enclosing_range 0:6-0:24
//      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ import scip-dotnet nuget . . CodeAnalysis/
//                                      enclosing_range 0:6-0:37
//             ^^^^^^^^^^^ read_access scip-dotnet nuget . . Diagnostics/
//                         enclosing_range 0:6-0:24
//                         ^^^^^^^^^^^^ read_access scip-dotnet nuget . . CodeAnalysis/
//                                      enclosing_range 0:6-0:37

  namespace Main;
//          ^^^^ read_access scip-dotnet nuget . . Main/
//               enclosing_range 2:10-2:14

  [SuppressMessage("ReSharper", "all")]
// ^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
//                 enclosing_range 4:1-4:16
  public class Interfaces
//             ^^^^^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#
//                        enclosing_range 4:0-69:1
//                        documentation ```cs\nclass Interfaces\n```
//                        kind Class
  {
      public interface IOne
//                     ^^^^ definition scip-dotnet nuget . . Main/Interfaces#IOne#
//                          enclosing_range 7:4-9:6
//                          documentation ```cs\ninterface IOne\n```
//                          kind Interface
      {
      };

      public interface ITwo
//                     ^^^^ definition scip-dotnet nuget . . Main/Interfaces#ITwo#
//                          enclosing_range 11:4-13:6
//                          documentation ```cs\ninterface ITwo\n```
//                          kind Interface
      {
      };

      public interface IThree
//                     ^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IThree#
//                            enclosing_range 15:4-17:6
//                            documentation ```cs\ninterface IThree\n```
//                            kind Interface
      {
      };

      public interface IProperties
//                     ^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IProperties#
//                                 enclosing_range 19:4-25:5
//                                 documentation ```cs\ninterface IProperties\n```
//                                 kind Interface
      {
          byte Get { get; }
//             ^^^ definition forward_definition scip-dotnet nuget . . Main/Interfaces#IProperties#Get.
//                 enclosing_range 21:8-21:25
//                 documentation ```cs\nbyte IProperties.Get { get; }\n```
//                 kind Property
//                 relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Byte#
          char Set { set; }
//             ^^^ definition forward_definition scip-dotnet nuget . . Main/Interfaces#IProperties#Set.
//                 enclosing_range 22:8-22:25
//                 documentation ```cs\nchar IProperties.Set { set; }\n```
//                 kind Property
//                 relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Char#
          uint GetSet { get; set; }
//             ^^^^^^ definition forward_definition scip-dotnet nuget . . Main/Interfaces#IProperties#GetSet.
//                    enclosing_range 23:8-23:33
//                    documentation ```cs\nuint IProperties.GetSet { get; set; }\n```
//                    kind Property
//                    relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/UInt32#
          long SetGet { set; get; }
//             ^^^^^^ definition forward_definition scip-dotnet nuget . . Main/Interfaces#IProperties#SetGet.
//                    enclosing_range 24:8-24:33
//                    documentation ```cs\nlong IProperties.SetGet { get; set; }\n```
//                    kind Property
//                    relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
      }

      interface IMethods
//              ^^^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IMethods#
//                       enclosing_range 27:4-33:6
//                       documentation ```cs\ninterface IMethods\n```
//                       kind Interface
      {
          void Nothing();
//             ^^^^^^^ definition forward_definition scip-dotnet nuget . . Main/Interfaces#IMethods#Nothing().
//                     enclosing_range 29:8-29:23
//                     documentation ```cs\nvoid IMethods.Nothing()\n```
//                     kind AbstractMethod
          int Output();
//            ^^^^^^ definition forward_definition scip-dotnet nuget . . Main/Interfaces#IMethods#Output().
//                   enclosing_range 30:8-30:21
//                   documentation ```cs\nint IMethods.Output()\n```
//                   kind AbstractMethod
          void Input(string a);
//             ^^^^^ definition forward_definition scip-dotnet nuget . . Main/Interfaces#IMethods#Input().
//                   enclosing_range 31:8-31:29
//                   documentation ```cs\nvoid IMethods.Input(string a)\n```
//                   kind AbstractMethod
//                          ^ definition scip-dotnet nuget . . Main/Interfaces#IMethods#Input().(a)
//                            enclosing_range 31:8-31:29
//                            documentation ```cs\nstring a\n```
//                            kind Parameter
//                            relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
          int InputOutput(string a);
//            ^^^^^^^^^^^ definition forward_definition scip-dotnet nuget . . Main/Interfaces#IMethods#InputOutput().
//                        enclosing_range 32:8-32:34
//                        documentation ```cs\nint IMethods.InputOutput(string a)\n```
//                        kind AbstractMethod
//                               ^ definition scip-dotnet nuget . . Main/Interfaces#IMethods#InputOutput().(a)
//                                 enclosing_range 32:8-32:34
//                                 documentation ```cs\nstring a\n```
//                                 kind Parameter
//                                 relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
      };

      interface IEvent
//              ^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IEvent#
//                     enclosing_range 35:4-38:5
//                     documentation ```cs\ninterface IEvent\n```
//                     kind Interface
      {
          event EventHandler<int> SomeEvent;
//                                ^^^^^^^^^ definition forward_definition scip-dotnet nuget . . Main/Interfaces#IEvent#SomeEvent#
//                                          enclosing_range 37:8-37:42
//                                          documentation ```cs\nevent EventHandler<int> IEvent.SomeEvent\n```
//                                          kind Event
//                                          relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/EventHandler#
      }

      interface IIndex
//              ^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IIndex#
//                     enclosing_range 40:4-43:5
//                     documentation ```cs\ninterface IIndex\n```
//                     kind Interface
      {
          bool this[int index] { get; set; }
//                      ^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IIndex#`this[]`.(index)
//                            enclosing_range 42:8-42:42
//                            documentation ```cs\nint index\n```
//                            kind Parameter
//                            relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
      }

      interface IDefault
//              ^^^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IDefault#
//                       enclosing_range 45:4-51:5
//                       documentation ```cs\ninterface IDefault\n```
//                       kind Interface
      {
          void Log(string message)
//             ^^^ definition scip-dotnet nuget . . Main/Interfaces#IDefault#Log().
//                 enclosing_range 47:8-50:9
//                 documentation ```cs\nvoid IDefault.Log(string message)\n```
//                 kind Method
//                        ^^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IDefault#Log().(message)
//                                enclosing_range 47:8-50:9
//                                documentation ```cs\nstring message\n```
//                                kind Parameter
//                                relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
          {
              Console.WriteLine(message);
//            ^^^^^^^ read_access scip-dotnet nuget System.Console 10.0.0.0 System/Console#
//                    enclosing_range 49:12-49:38
//                    ^^^^^^^^^ read_access scip-dotnet nuget System.Console 10.0.0.0 System/Console#WriteLine(+11).
//                              enclosing_range 49:12-49:38
//                              ^^^^^^^ read_access scip-dotnet nuget . . Main/Interfaces#IDefault#Log().(message)
//                                      enclosing_range 49:30-49:37
          }
      }


      private interface IInherit : IOne, ITwo
//                      ^^^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IInherit#
//                               enclosing_range 54:4-56:5
//                               documentation ```cs\ninterface IInherit\n```
//                               kind Interface
//                               relationship implementation scip-dotnet nuget . . Main/Interfaces#IOne#
//                               relationship implementation scip-dotnet nuget . . Main/Interfaces#ITwo#
//                                 ^^^^ read_access scip-dotnet nuget . . Main/Interfaces#IOne#
//                                      enclosing_range 54:33-54:37
//                                       ^^^^ read_access scip-dotnet nuget . . Main/Interfaces#ITwo#
//                                            enclosing_range 54:39-54:43
      {
      }

      public interface IGetNext<T> where T : IGetNext<T>
//                     ^^^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IGetNext#
//                              enclosing_range 58:4-64:5
//                              documentation ```cs\ninterface IGetNext<T> where T : IGetNext<T>\n```
//                              kind Interface
//                              ^ definition local 0
//                                enclosing_range 58:4-64:5
//                                documentation ```cs\nT\n```
//                                kind TypeParameter
//                                       ^ read_access local 0
//                                         enclosing_range 58:39-58:40
//                                                    ^ read_access local 0
//                                                      enclosing_range 58:52-58:53
      {
          static IGetNext<T> operator ++(IGetNext<T> other)
//                        ^ read_access local 0
//                          enclosing_range 60:24-60:25
//                                                ^ read_access local 0
//                                                  enclosing_range 60:48-60:49
//                                                   ^^^^^ definition scip-dotnet nuget . . Main/Interfaces#IGetNext#op_Increment().(other)
//                                                         enclosing_range 60:8-63:9
//                                                         documentation ```cs\nIGetNext<T> other\n```
//                                                         kind Parameter
//                                                         relationship type_definition scip-dotnet nuget . . Main/Interfaces#IGetNext#
          {
              throw new NotImplementedException();
//                      ^^^^^^^^^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 System/NotImplementedException#
//                                              enclosing_range 62:18-62:47
          }
      }

      private interface ITypeParameter<T1, T2> : ITwo where T1 : IOne where T2 : IThree
//                      ^^^^^^^^^^^^^^ definition scip-dotnet nuget . . Main/Interfaces#ITypeParameter#
//                                     enclosing_range 66:4-68:5
//                                     documentation ```cs\ninterface ITypeParameter<T1, T2> where T1 : IOne where T2 : IThree\n```
//                                     kind Interface
//                                     relationship implementation scip-dotnet nuget . . Main/Interfaces#ITwo#
//                                     ^^ definition local 1
//                                        enclosing_range 66:4-68:5
//                                        documentation ```cs\nT1\n```
//                                        kind TypeParameter
//                                         ^^ definition local 2
//                                            enclosing_range 66:4-68:5
//                                            documentation ```cs\nT2\n```
//                                            kind TypeParameter
//                                               ^^^^ read_access scip-dotnet nuget . . Main/Interfaces#ITwo#
//                                                    enclosing_range 66:47-66:51
//                                                          ^^ read_access local 1
//                                                             enclosing_range 66:58-66:60
//                                                               ^^^^ read_access scip-dotnet nuget . . Main/Interfaces#IOne#
//                                                                    enclosing_range 66:63-66:67
//                                                                          ^^ read_access local 2
//                                                                             enclosing_range 66:74-66:76
//                                                                               ^^^^^^ read_access scip-dotnet nuget . . Main/Interfaces#IThree#
//                                                                                      enclosing_range 66:79-66:85
      {
      }
  }

File: snapshots/output-net10.0/syntax/VBMain/Interfaces.vb
  Imports System.Diagnostics.CodeAnalysis
'         ^^^^^^ read_access scip-dotnet nuget . . System/
'                enclosing_range 0:8-0:26
'         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ import scip-dotnet nuget . . CodeAnalysis/
'                                         enclosing_range 0:8-0:39
'                ^^^^^^^^^^^ read_access scip-dotnet nuget . . Diagnostics/
'                            enclosing_range 0:8-0:26
'                            ^^^^^^^^^^^^ read_access scip-dotnet nuget . . CodeAnalysis/
'                                         enclosing_range 0:8-0:39

  Namespace VBMain
'           ^^^^^^ read_access scip-dotnet nuget . . VBMain/
'                  enclosing_range 2:10-2:16
      <SuppressMessage("ReSharper", "all")>
'      ^^^^^^^^^^^^^^^ read_access scip-dotnet nuget System.Runtime 10.0.0.0 CodeAnalysis/SuppressMessageAttribute#`.ctor`().
'                      enclosing_range 3:5-3:20
      Public Class Interfaces
'                  ^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#
'                             enclosing_range 3:4-4:27
'                             documentation ```vb\nClass Interfaces\n```
'                             kind Class
          Interface IOne
'                   ^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#IOne#
'                        enclosing_range 5:8-5:22
'                        documentation ```vb\nInterface IOne\n```
'                        kind Interface
          End Interface

          Interface ITwo
'                   ^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#ITwo#
'                        enclosing_range 8:8-8:22
'                        documentation ```vb\nInterface ITwo\n```
'                        kind Interface
          End Interface

          Interface IThree
'                   ^^^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#IThree#
'                          enclosing_range 11:8-11:24
'                          documentation ```vb\nInterface IThree\n```
'                          kind Interface
          End Interface

          Interface IProperties
'                   ^^^^^^^^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#IProperties#
'                               enclosing_range 14:8-14:29
'                               documentation ```vb\nInterface IProperties\n```
'                               kind Interface
              ReadOnly Property [Get] As Byte
'                               ^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IProperties#Get.
'                                     enclosing_range 15:12-15:43
'                                     documentation ```vb\nReadOnly Property IProperties.Get As Byte\n```
'                                     kind Property
'                                     relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Byte#
              WriteOnly Property [Set] As Char
'                                ^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IProperties#Set.
'                                      enclosing_range 16:12-16:44
'                                      documentation ```vb\nWriteOnly Property IProperties.Set As Char\n```
'                                      kind Property
'                                      relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Char#
              Property GetSet As UInteger
'                      ^^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IProperties#GetSet.
'                             enclosing_range 17:12-17:39
'                             documentation ```vb\nProperty IProperties.GetSet As UInteger\n```
'                             kind Property
'                             relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/UInt32#
              Property SetGet As Long
'                      ^^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IProperties#SetGet.
'                             enclosing_range 18:12-18:35
'                             documentation ```vb\nProperty IProperties.SetGet As Long\n```
'                             kind Property
'                             relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int64#
          End Interface

          Interface IMethods
'                   ^^^^^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#IMethods#
'                            enclosing_range 21:8-21:26
'                            documentation ```vb\nInterface IMethods\n```
'                            kind Interface
              Sub [Nothing]()
'                 ^^^^^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IMethods#Nothing().
'                           enclosing_range 22:12-22:27
'                           documentation ```vb\nSub IMethods.Nothing()\n```
'                           kind AbstractMethod
              Function Output() As Integer
'                      ^^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IMethods#Output().
'                             enclosing_range 23:12-23:40
'                             documentation ```vb\nFunction IMethods.Output() As Integer\n```
'                             kind AbstractMethod
              Sub Input(ByVal a As String)
'                 ^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IMethods#Input().
'                       enclosing_range 24:12-24:40
'                       documentation ```vb\nSub IMethods.Input(a As String)\n```
'                       kind AbstractMethod
'                             ^ definition scip-dotnet nuget . . VBMain/Interfaces#IMethods#Input().(a)
'                               enclosing_range 24:12-24:40
'                               documentation ```vb\na As String\n```
'                               kind Parameter
'                               relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
              Function InputOutput(ByVal a As String) As Integer
'                      ^^^^^^^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IMethods#InputOutput().
'                                  enclosing_range 25:12-25:62
'                                  documentation ```vb\nFunction IMethods.InputOutput(a As String) As Integer\n```
'                                  kind AbstractMethod
'                                        ^ definition scip-dotnet nuget . . VBMain/Interfaces#IMethods#InputOutput().(a)
'                                          enclosing_range 25:12-25:62
'                                          documentation ```vb\na As String\n```
'                                          kind Parameter
'                                          relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/String#
          End Interface

          Interface IEvent
'                   ^^^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#IEvent#
'                          enclosing_range 28:8-28:24
'                          documentation ```vb\nInterface IEvent\n```
'                          kind Interface
              Event SomeEvent As EventHandler(Of Integer)
'                   ^^^^^^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IEvent#SomeEvent#
'                             enclosing_range 29:12-29:55
'                             documentation ```vb\nEvent IEvent.SomeEvent As EventHandler(Of Integer)\n```
'                             kind Event
'                             relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/EventHandler#
          End Interface

          Interface IIndex
'                   ^^^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#IIndex#
'                          enclosing_range 32:8-32:24
'                          documentation ```vb\nInterface IIndex\n```
'                          kind Interface
              Default Property Item(ByVal index As Integer) As Boolean
'                              ^^^^ definition forward_definition scip-dotnet nuget . . VBMain/Interfaces#IIndex#Item.
'                                   enclosing_range 33:12-33:68
'                                   documentation ```vb\nDefault Property IIndex.Item(index As Integer) As Boolean\n```
'                                   kind Property
'                                   relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Boolean#
'                                         ^^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#IIndex#Item.(index)
'                                               enclosing_range 33:12-33:68
'                                               documentation ```vb\nindex As Integer\n```
'                                               kind Parameter
'                                               relationship type_definition scip-dotnet nuget System.Runtime 10.0.0.0 System/Int32#
          End Interface

          Private Interface IInherit
'                           ^^^^^^^^ definition scip-dotnet nuget . . VBMain/Interfaces#IInherit#
'                                    enclosing_range 36:8-36:34
'                                    documentation ```vb\nInterface IInherit\n```
'                                    kind Interface
'                                    relationship implementation scip-dotnet nuget . . VBMain/Interfaces#IOne#
'                                    relationship implementation scip-dotnet nuget . . VBMain/Interfaces#ITwo#
              Inherits IOne, ITwo
'                      ^^^^ read_access scip-dotnet nuget . . VBMain/Interfaces#IOne#
'                           enclosing_range 37:21-37:25
'                            ^^^^ read_access scip-dotnet nuget . . VBMain/Interfaces#ITwo#
'                                 enclosing_range 37:27-37:31
          End Interface
      End Class
  End Namespace
